hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlSerializerHook.java

```diff
@@ -117,6 +272,103 @@ public DataSerializableFactory createFactory() {
+        constructors[EXPRESSION_POSITION] = arg -> new PositionFunction();
+        constructors[EXPRESSION_REMAINDER] = arg -> new RemainderFunction<>();
+        constructors[EXPRESSION_CONCAT_WS] = arg -> new ConcatWSFunction();
+        constructors[EXPRESSION_CASE] = arg -> new CaseExpression<>();
+        constructors[EXPRESSION_EXTRACT] = arg -> new ExtractFunction();
+
+        constructors[EXPRESSION_DOUBLE_DOUBLE] = arg -> new DoubleBiFunction();
+        constructors[EXPRESSION_TO_TIMESTAMP_TZ] = arg -> new ToTimestampTzFunction();
+        constructors[EXPRESSION_TO_EPOCH_MILLIS] = arg -> new ToEpochMillisFunction();
+
+        constructors[SARG_EXPRESSION] = arg -> new SargExpression<>();
+        constructors[SEARCH_PREDICATE] = arg -> new SearchPredicate();
+        constructors[EXPRESSION_FIELD_ACCESS] = arg -> new FieldAccessExpression<>();
+        constructors[EXPRESSION_ROW] = arg -> new RowExpression();
+        constructors[MAP_FETCH_INDEX_OPERATION] = arg -> new MapFetchIndexOperation();
+        constructors[INDEX_ITERATION_POINTER] = arg -> new IndexIterationPointer();
+        constructors[MAP_FETCH_INDEX_OPERATION_RESULT] = arg -> new MapFetchIndexOperationResult();
+
+        constructors[DATA_LINK] = arg -> new DataLinkCatalogEntry();
+
+        constructors[QUERY_DATA_TYPE] = arg -> new QueryDataType();
+
+        constructors[QUERY_ID] = arg -> new QueryId();
+
+        constructors[ROW_HEAP] = arg -> new HeapRow();
+        constructors[ROW_EMPTY] = arg -> EmptyRow.INSTANCE;
+
+        constructors[LAZY_TARGET] = arg -> new LazyTarget();
+
+        constructors[TARGET_DESCRIPTOR_GENERIC] = arg -> GenericQueryTargetDescriptor.DEFAULT;
+
+        constructors[QUERY_PATH] = arg -> new QueryPath();
+
+        constructors[INTERVAL_YEAR_MONTH] = arg -> new SqlYearMonthInterval();
+        constructors[INTERVAL_DAY_SECOND] = arg -> new SqlDaySecondInterval();
+
+        constructors[QUERY_DATA_TYPE_FIELD] = arg -> new QueryDataType.QueryDataTypeField();
+
         return new ArrayDataSerializableFactory(constructors);
     }
+
+    @Override
+    public void afterFactoriesCreated(Map<Integer, DataSerializableFactory> factories) {
+        ArrayDataSerializableFactory mapDataFactory = (ArrayDataSerializableFactory) factories.get(SqlDataSerializerHook.F_ID);
+
+        ConstructorFunction<Integer, IdentifiedDataSerializable>[] constructors =
+                new ConstructorFunction[SqlDataSerializerHook.TYPE_FIELD + 1];
+        constructors[SqlDataSerializerHook.MAPPING] = arg -> new Mapping();
+        constructors[SqlDataSerializerHook.MAPPING_FIELD] = arg -> new MappingField();
+        constructors[SqlDataSerializerHook.VIEW] = arg -> new View();
+        constructors[SqlDataSerializerHook.TYPE] = arg -> new Type();
+        constructors[SqlDataSerializerHook.TYPE_FIELD] = arg -> new Type.TypeField();
+
+        mapDataFactory.mergeConstructors(constructors);
+    }
```

Comment:
I don't feel great about overriding a different hook's contents here. I see that these classes are defined in the hazelcast-sql module. Why do we need to have them in the SqlDataSerializerHook?